src/BonVoyage/Clients/VenuesClient.cs

```diff
@@ -49,6 +49,32 @@ public async Task<IReadOnlyCollection<CompactVenue>> Search(string placeName, st
                 var jObject = DeserializeObject<JObject>(resultAsString);
                 var venues = DeserializeObject<IEnumerable<CompactVenue>>(jObject["response"]["venues"].ToString());
 
+                return new ReadOnlyCollection<CompactVenue>(venues.ToList());
+            }
+        }
+        /// <param name="location">
+        /// The location coordinates (latitude and longitude) in the world (i.e. 37.77,-122.41). Used to pass as the value for 'll' parameter.
+        /// </param>
+        /// <param name="categoryId">
+        /// The id of the category to limit results to. If specifying a top-level category, all sub-categories will also match the query.
+        /// </param>
+        /// <param name="limit">The number of search results. Min: 1, Max: 50</param>
+        /// <seealso href="https://developer.foursquare.com/docs/venues/search" />
+        public async Task<IReadOnlyCollection<CompactVenue>> Search(Location location, string categoryId, int limit)
+        {
+            if (location == null) throw new ArgumentNullException(nameof(location));
+            if (categoryId == null) throw new ArgumentNullException(nameof(categoryId));
+            if (limit < 1) throw new ArgumentOutOfRangeException(nameof(limit), limit, "Cannot be lower than 1");
+            if (limit > 50) throw new ArgumentOutOfRangeException(nameof(limit), limit, "Cannot be greater than 50");
+
+            NumberFormatInfo nfi = new NumberFormatInfo();
```

Comment:
do you need this if you were to pass `CultureInfo.InvariantCulture` to `ToString` call?

Reply:
I just copied the request that was already in the code, and added the `ll` parameter. The part with `CultureInfo.InvariantCulture` was already there in the `        public async Task<IReadOnlyCollection<CompactVenue>> Search(string placeName, string categoryId, int limit)` method. I didn't try removing it, because i once had a problem with number formatting using the foursquare API